angular/projects/admin-nrpti/src/app/communications/communications.component.ts

```diff
@@ -0,0 +1,113 @@
+  styleUrls: ['./communications.component.scss']
+})
+export class CommunicationsComponent implements OnInit, OnDestroy {
+  @ViewChild(DatePickerComponent) DatePicker: DatePickerComponent;
+
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  public myForm: FormGroup;
+
+  public commPackage: CommunicationsPackage;
+  public selectedApplication: string;
+
+  constructor(
+    public route: ActivatedRoute,
+    public router: Router,
+    private factoryService: FactoryService,
+    private utils: Utils,
+    private loadingScreenService: LoadingScreenService
+  ) {}
+
+  ngOnInit() {
+    this.route.data.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res: any) => {
+      this.commPackage = res.communicationsPackage.COMMUNICATIONS;
+      const urlSegments = this.router.url.split('/');
+      this.selectedApplication = urlSegments[urlSegments.length - 1].toUpperCase();
+
+      this.buildForm();
+    });
+  }
+
+  private buildForm() {
+    this.myForm = new FormGroup({
+      popupTitle: new FormControl({
+        value: (this.commPackage && this.commPackage.title) || '',
+        disabled: !this.checkRoles()
+      }),
+      description: new FormControl({
+        value: (this.commPackage && this.commPackage.description) || '',
+        disabled: !this.checkRoles()
+      }),
+      startDate: new FormControl({
+        value: (this.commPackage &&
+          this.commPackage.startDate &&
+          this.utils.convertJSDateToNGBDate(new Date(this.commPackage.startDate))) ||
+        '',
+        disabled: !this.checkRoles()
+      }),
+      endDate: new FormControl({
+        value: (this.commPackage &&
+          this.commPackage.endDate &&
+          this.utils.convertJSDateToNGBDate(new Date(this.commPackage.endDate))) ||
+        '',
+        disabled: !this.checkRoles()
+      })
+    });
+  }
+
+  async submit() {
+    // save the popup
+    this.loadingScreenService.setLoadingState(true, 'main');
+
+    if (
+      !this.myForm.get('popupTitle').value ||
+      !this.myForm.get('description').value
+    ) {
+      alert('Please ensure your Communication Package includes a title and a description.');
+      return;
+    }
+
+    const popup = {};
+
+    popup['title'] = this.myForm.get('popupTitle').value;
+    popup['description'] = this.myForm.get('description').value;
+    popup['startDate'] = this.utils.convertFormGroupNGBDateToJSDate(this.myForm.get('startDate').value);
+    popup['endDate'] = this.utils.convertFormGroupNGBDateToJSDate(this.myForm.get('endDate').value);
+    popup['application'] = this.selectedApplication;
+
+    this.factoryService.createCommunicationPackage(popup).subscribe(async (res: any) => {
+      this.loadingScreenService.setLoadingState(false, 'main');
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  checkRoles() {
+    if (this.factoryService.userInRole('sysadmin')) {
```

Comment:
I realize this was not added in this PR, but these user helper functions don't really belong in a factory service, do they?

Reply:
Yeah, I think they're there because factory service had the imports in place for the keycloak service, etc. We can certainly move them... probably a good tech-debt ticket for future enhancement as I don't want to alter all the places where the `isIn<app>Role()` functions are called with this ticket particularly.

Reply:
Yeah, not a worry for this ticket.